### ISSUE TITLE
Adding Site Channel: Missing way to correct "Channel could not be verified!"

### ISSUE THREAD

**Issue by lgebhardt** [brave-intl/publishers#711]
Since #674/#655 if the user enters the token wrong they will see "Channel could not be verified!" with no way to "reenter" the channel to either choose a new method or check the instructions or token:

![screen shot 2018-03-12 at 10 59 22 am](https://user-images.githubusercontent.com/29154/37291388-e5a45f54-25e4-11e8-9c1e-febfe08289c4.png)

I propose we restore the previous behavior where clicking the channel will take the user to viewing the channel where they last left off (when they clicked verify).

From #655: "When an error has been encountered during domain verification, display that error for the channel and provide actionable steps. These scenarios will be spelled out further in a separate issue." I'm not sure how this is to be implemented, but it could provide an alternative way for the user to correct the problem instead of the above solution.

**Comment by nvonpentz** [comment]
@jenn-rhim can you please comment?

**Comment by jenn-rhim** [comment]
Technically, 'verification failed' is not a valid state to display because channels on the dashboard are supposed to be either 'verified' or 'pending'. or else shouldn't be on the dashboard, which serves no purpose. 

I do agree that it'd be nice to offer a chance to retry! 

1. The below mockup shows the dialog to handle this case. We do have a template handling a similar case for YT channel (claimed/unavailable). 

2. The difference is the content, and if we were to do it right, be helpful, the below bullet points would have to be process specific (i.e. regular website/wordpress/etc. + method chosen).

3. The time to display this dialog is when the system returns the results, which could be instant after clicking the Verify button from the last step.

4. When the user chooses to retry, we send them back to the first step (thought about just re-triggering the last step, but probably not too useful, not to mention the odds of producing a different results would be low).

5. When the user dismisses the dialog, the channel should transition out (any animation suggestions?), if it was on the dashboard as a pending state previously.  This will result in shuffling the channel blocks a bit if multiple.

6. If this was adding the brand new channel, user will basically return to the dashboard in the same state as before starting the verification process.

![verification-failed](https://user-images.githubusercontent.com/4369856/37498017-004a0156-2879-11e8-9379-ee93d3c42368.png)

**Comment by nvonpentz** [comment]
I suggest we actually link them to the file location so they can check themselves.
```
{publisher-website.com}/.well-known/brave-payments-verification.txt
```

I'm not sure what info eyeshade gives us right now but it would be useful if we could tell the user if their:

* public file isn't there or is in the wrong place `404`
* public file is hidden `403`
* token is incorrect (public file & dns)
* there is an unknown problem (please contact publisher+support@basicattentiontoken.org)

We should also send emails when a verification has failed, but that might be outside the scope of this issue.

**Comment by jenn-rhim** [comment]
@nvonpentz 
I like your idea of providing the link -- definitely should list all troubleshoot cases you have there (and possibly more) before the link though (meaning before they click the link). They can always come back from the tab with the file location but better if they know what to expect.   as far as i know, eyeshade is very passive-- it only responds to our call and doesn't return any data in terms of the failure causes -- should investigate on improving this (@mrose17 )

**Comment by nvonpentz** [comment]
@jenn-rhim 
* Should we also email them the failed verification information? If we choose to email them should we also display this dialog?
* If they dismiss the dialog will they be able to open it up again? (I think they should)
* Will be display the verification token on dialog or only on the 'let's finish' page?
* Do we want to consider combining the 'let's finish' and this dialog into a single page?

@lgebhardt @mrose17 
Where are we on eyeshade sending back information about why a verification failed?

**Comment by jenn-rhim** [comment]
Nick 
- yes, we should send out an email. 
- no, there won’t t be any triggering point as we remove the channel if the user clicks ‘ok’. Remember that restarting this process is not that costly. Most errors are likely out of bound steps so continue providing this keeping the UI element just increase the management cost. It’d be a better approach to identify where people tend to get stuck. I’m running some user survey. 
- that’s a mid term plan ‘if’ we get more meaningful response from eyeshade as to what the cause of the failure is/might be. Some users may not be familiar with the notion of tokens so I’d rather be specific than providing it to everyone. 
- I don’t understand the question. those are 2 different use cases: one is pause and the other one is failed. Let me know what you’re trying to accomplish by suggesting this.
